model/data/mailbox.h

```diff
@@ -19,24 +22,15 @@ class Mailbox {
 public:
     Mailbox(Impl impl) : impl(std::move(impl)) {}
 
-    /**
-     * Handler wrapper specifies handler signature and default values for
-     * no message or error.
-     */
     template <typename Handler>
-    struct OnMessage {
-        Handler h;
-        void operator() (error_code e, optional<Message> m = optional<Message>()) {
```

Comment:
The matter of OnMessage existence is default value for m and implicit handler signature :) So, it can be called as h(e) for error handling everywhere - what is pretty convenient. :)